Firestore/core/src/util/byte_stream_apple.mm

```diff
@@ -26,23 +26,21 @@
 namespace firestore {
 namespace util {
 
-StreamReadResult ByteStreamNSInputStream::ReadUntil(char delim,
-                                                    size_t max_length) {
+StreamReadResult ByteStreamApple::ReadUntil(char delim, size_t max_length) {
   size_t found_at = std::string::npos;
-  do {
-    found_at = buffer_.find(delim);
-    if (found_at != std::string::npos) {
-      break;
-    }
+  size_t last_pos = 0;
+  while (buffer_.size() < max_length) {
     auto read = ReadToBuffer(max_length - buffer_.size());
     if (read < 0) {
       return ErrorResult();
     }
-    if (read == 0) {
+
+    found_at = buffer_.find(delim, last_pos);
+    last_pos = buffer_.size();
+    if (found_at != std::string::npos || read == 0) {
       break;
     }
-
-  } while (buffer_.size() < max_length);
+  }
 
   // One last try since the loop might break because eof or max_length reached.
```

Comment:
Question: is this check still necessary? If yes, can you add a test that fails without it?